tests/integration/test_wpk/test_wpk_agent.py

```diff
@@ -0,0 +1,272 @@
+# Copyright (C) 2015-2020, Wazuh Inc.
+# Created by Wazuh, Inc. <info@wazuh.com>.
+# This program is free software; you can redistribute it and/or modify it under the terms of GPLv2
+import hashlib
+import os
+import platform
+import pytest
+import time
+import requests
+import subprocess
+import yaml
+
+from configobj import ConfigObj
+from datetime import datetime
+from wazuh_testing.tools import WAZUH_PATH
+from wazuh_testing.tools.authd_sim import AuthdSimulator
+from wazuh_testing.tools.configuration import load_wazuh_configurations
+from wazuh_testing.tools.file import truncate_file
+from wazuh_testing.tools.remoted_sim import RemotedSimulator
+from wazuh_testing.tools.services import control_service
+
+pytestmark = [pytest.mark.tier(level=0), pytest.mark.agent]
+
+folder = 'etc' if platform.system() == 'Linux' else ''
+
+DEFAULT_UPGRADE_SCRIPT = 'upgrade.sh' if platform.system() == 'Linux' else 'upgrade.bat'
+CLIENT_KEYS_PATH = os.path.join(WAZUH_PATH, folder,'client.keys') #for unix add 'etc'
+CRYPTO = "aes"
+SERVER_ADDRESS = 'localhost'
+PROTOCOL = "tcp"
+
+# Test will varying according to agent version. This test should be tried with at least:
+# 1. v3.13.1
+# 2. v4.0.0
+config_file_path = os.path.join(WAZUH_PATH, 'etc', 'ossec-init.conf')
+_config = ConfigObj(config_file_path)
+_agent_version = _config['VERSION']
+
+test_metadata = [
+    # 1. Upgrade from initial_version to v4.0.0
+    {
+        'protocol': PROTOCOL,
+        'initial_version': _agent_version,
+        'agent_version': 'v4.0.0',
```

Comment:
Maybe the agent_version should be a variable. This way, we will be able to test with different versions like 4.1, etc